tx/tx_dynamic_fee.go

```diff
@@ -0,0 +1,127 @@
+// Copyright (c) 2024 The VeChainThor developers
+
+// Distributed under the GNU Lesser General Public License v3.0 software license, see the accompanying
+// file LICENSE or <https://www.gnu.org/licenses/lgpl-3.0.html>
+
+package tx
+
+import (
+	"io"
+	"math"
+	"math/big"
+
+	"github.com/ethereum/go-ethereum/rlp"
+	"github.com/vechain/thor/v2/thor"
+)
+
+type DynamicFeeTransaction struct {
```

Comment:
Maybe we should name this `Transaction` and name the old one `LegacyTransaction`

I think its more in line with what they do in Geth